src/wp-includes/class-wp-oembed.php

```diff
@@ -414,6 +414,11 @@ public function get_html( $url, $args = '' ) {
 			return false;
 		}
 
+		// Lazy load oEmbed iframes.
+		if ( wp_lazy_loading_enabled( 'iframe', 'oembed' ) && isset( $data->html ) && str_starts_with( $data->html, '<iframe' ) && ! str_contains( $data->html, 'loading' ) ) {
```

Comment:
`str_starts_with` is not 100% accurate because there could be other HTML before the iframe

Reply:
Probably better to use `WP_HTML_Tag_Processor` to check for the element existing and having the `loading` attribute.

Reply:
This should do it:

```php
$p = new WP_HTML_Tag_Processor( $html );
if ( $p->next_tag( 'iframe' ) && ! $p->get_attribute( 'loading' ) ) {
    $html = wp_iframe_tag_add_loading_attr( $html, 'oembed' );
}
```

Note that `wp_iframe_tag_add_loading_attr` should also be updated to use `WP_HTML_Tag_Processor` as there is already a TODO to do so.

Reply:
updated